src/python/pants/backend/codegen/protobuf/python/python_protobuf_subsystem.py

```diff
@@ -53,6 +53,16 @@ def register_options(cls, register):
                 "`--mypy-plugin` for this option to be used."
             ),
         )
+        register(
+            "--dependency-inference",
+            type=bool,
+            default=True,
+            advanced=True,
+            help=(
+                "Infer when Python code depends on generated Protobuf files. This requires that "
+                "`[python-infer].imports` is also true (the default)."
+            ),
+        )
```

Comment:
Given that the `[python-infer].imports` flag exists, and the first party mapping is lazily created only if inference is turned on, is it necessary to have this flag as well? IMO, would avoid it until someone indicates they need it.

Reply:
Sure, that sounds reasonable. I'm generally in favor of fewer options.